utils/abi-coder.js

```diff
@@ -971,6 +971,19 @@ utils.defineProperty(Coder.prototype, 'encode', function(names, types, values) {
     return utils.hexlify(coderTuple(this.coerceFunc, coders).encode(values));
 });
 
+utils.defineProperty(Coder.prototype, 'decodeRevertReason', function(data) {
+
+    data = utils.arrayify(data);
+
+    if (data && utils.hexlify(data.slice(0, 4)) === '0x08c379a0') {
+        var result = coderTuple(this.coerceFunc, [
+            getParamCoder(this.coerceFunc, {type: 'string'}, undefined),
+        ]).decode(data.slice(4), 0);
+
+        return result.value[0];
```

Comment:
You need to have a check on the result. If it does not have .value or .value is not array you are fucked